src/lib.rs

```diff
@@ -277,21 +277,26 @@ pub struct Cmd {
     args: Vec<OsString>,
     stdin_contents: Option<Vec<u8>>,
     ignore_status: bool,
+    echo_cmd: bool,
 }
 
 impl fmt::Display for Cmd {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        let mut space = "";
-        for arg in &self.args {
-            write!(f, "{}", space)?;
-            space = " ";
+        if self.echo_cmd {
+            let mut space = "";
+            for arg in &self.args {
+                write!(f, "{}", space)?;
+                space = " ";
 
-            let arg = arg.to_string_lossy();
-            if arg.chars().any(|it| it.is_ascii_whitespace()) {
-                write!(f, "\"{}\"", arg.escape_default())?
-            } else {
-                write!(f, "{}", arg)?
-            };
+                let arg = arg.to_string_lossy();
+                if arg.chars().any(|it| it.is_ascii_whitespace()) {
+                    write!(f, "\"{}\"", arg.escape_default())?
+                } else {
+                    write!(f, "{}", arg)?
+                };
+            }
+        } else {
+            write!(f, "<hidden>")?;
```

Comment:
Do we need this case though? I think it's fair to assume that, if the command is secrent, you aren't going to print it. OTOH, I think it might be useful to print a command with `echo_cmd: false` sometimes (for example, in some error log)

Reply:
I added this because the command was still being printed if it failed. My original goal was to hide a command that had sensitive info in its command line, and for that you don't even want to print it in an error log, IMO. It seemed more sensible to hide the info here rather than having to check that flag in at least two places.

I think not echoing the command (because it's just not interesting or so) is a slightly different use case. I suppose I could split this up into two flags?

Reply:
Yeah, I agree with your analysis, two flags would work better here!

Reply:
Ok, I added another flag. 😃 I'm open to alternative suggestions on the name; I don't think it's terrible, but I'm not too attached to it either.